### ISSUE TITLE
getVersionsForView API doesn't return unpublished version

### ISSUE THREAD

**Issue by alansemenov** [enonic/xp#8314]
Discovered while working on [this case](https://github.com/enonic/app-contentstudio/issues/2014) (initiated by customer complaints) 

After content is unpublished, the version that was published is no longer shown in the Version History widget (it's not in response from `getVersionsForView` API).

Here's how response and UI look for a published version. We know that the content is currently published because `workspaces` has "_master_" and there's `publishInfo` in response where we can take date/time stamp from to build the UI.

![Screenshot 2020-08-31 at 12 26 46](https://user-images.githubusercontent.com/11612490/91710953-5e178980-eb85-11ea-8a0d-4aa7167fec3f.png)

![image](https://user-images.githubusercontent.com/11612490/91710244-3e339600-eb84-11ea-97d7-c8dd7268ceb3.png)

Here's the same content after unpublishing:

![Screenshot 2020-08-31 at 12 35 46](https://user-images.githubusercontent.com/11612490/91711816-d9c60600-eb86-11ea-8d6a-ade2d733168a.png)

![image](https://user-images.githubusercontent.com/11612490/91711510-4ee50b80-eb86-11ea-994b-24e7387fa624.png)

There's still just one version, which is now missing `publishInfo` and only has "_draft_" in `workspaces`.

So, basically there's no way of knowing:
1) When, by whom and for how long the content was published
2) When and by whom it was unpublished

As far as I see, there are **two problems** here:
1. We don't create a new version when a content is published. This is the version that should be kept in the draft branch after it's deleted from the master.
2. We don't create a new version when a content is unpublished.

Without solving these two we cannot build a realistic log after content was unpublished.

**Comment by sigdestad** [comment]
Hmm.. this is at least somewhat incorrect.

1. We always create a new version the first time content is published (and then we set published.first)
2. This is correct, and we probably should create a version when content is unpublished as well.

Finally, there is a filter that hides version with same timestamp (or similar) from being in the version list. It is kind of a hack, but has a reason, and is probably why you don't see the specific version

**Comment by alansemenov** [comment]
I'm not sure what exactly happens there - we have to figure this out - but the fact remains: there's one version returned by version API after the content is **published** and there's still just one version after the content is **unpublished** (and it's not the same version as the first one).

**Comment by sigdestad** [comment]
Ah, i think we figured out it created new version on unpublish, but did not change any data

**Comment by rymsha** [comment]
> Ah, i think we figured out it created new version on unpublish, but did not change any data
This is not fully enough to explain the behaviour, but with "deduplication" by last_modified timestamp value everything gets clear.
Current idea 
1. Always consider commited versions significant and never deduplicate them.
2. Commit unpublished version. (So 1 will be effective)
3. Don't nullify publishInfo on unpublished version. Instead return modifier, modified and timestamp values related to `unpublish` event.

**Comment by sigdestad** [comment]
The de-duplication was initially made because we were listing two similar items in the Version history, but maybe the UI should be in charge if it wants to drop listing one or more specific versions?
